fluss-server/src/main/java/org/apache/fluss/server/coordinator/CoordinatorEventProcessor.java

```diff
@@ -1062,6 +1073,71 @@ private CommitLakeTableSnapshotResponse tryProcessCommitLakeTableSnapshot(
+
+        // TODO here we need to check tabletServerEpoch, avoid to receive controlled shutdown
+        // request from an old tabletServer. Trace by https://github.com/alibaba/fluss/issues/1153
+        int tabletServerEpoch = controlledShutdownEvent.getTabletServerEpoch();
+
+        int tabletServerId = controlledShutdownEvent.getTabletServerId();
+        LOG.info(
+                "Try to process controlled shutdown for tabletServer: {} of tabletServer epoch: {}",
+                controlledShutdownEvent.getTabletServerId(),
+                tabletServerEpoch);
+
+        if (!coordinatorContext.liveOrShuttingDownTabletServers().contains(tabletServerId)) {
+            throw new TabletServerNotAvailableException(
+                    "TabletServer" + tabletServerId + " is not available.");
+        }
+
+        coordinatorContext.shuttingDownTabletServers().add(tabletServerId);
+        LOG.debug(
+                "All shutting down tabletServers: {}",
+                coordinatorContext.shuttingDownTabletServers());
+        LOG.debug("All live tabletServers: {}", coordinatorContext.liveTabletServerSet());
+
+        List<TableBucketReplica> replicasToActOn =
+                coordinatorContext.replicasOnTabletServer(tabletServerId).stream()
+                        .filter(
+                                replica -> {
+                                    TableBucket tableBucket = replica.getTableBucket();
+                                    return !coordinatorContext.getAssignment(tableBucket).isEmpty()
+                                            && coordinatorContext
+                                                    .getBucketLeaderAndIsr(tableBucket)
+                                                    .isPresent()
+                                            && !coordinatorContext.isToBeDeleted(tableBucket);
+                                })
+                        .collect(Collectors.toList());
+
+        Set<TableBucket> bucketsLedByServer = new HashSet<>();
+        Set<TableBucketReplica> replicasFollowedByServer = new HashSet<>();
+        for (TableBucketReplica replica : replicasToActOn) {
+            TableBucket tableBucket = replica.getTableBucket();
+            if (replica.getReplica()
+                    == coordinatorContext.getBucketLeaderAndIsr(tableBucket).get().leader()) {
+                bucketsLedByServer.add(tableBucket);
+            } else {
+                replicasFollowedByServer.add(replica);
+            }
+        }
+
+        tableBucketStateMachine.handleStateChange(
+                bucketsLedByServer, OnlineBucket, CONTROLLED_SHUTDOWN_ELECTION);
+
+        // TODO need send stop request to the leader?
+
+        // If the tabletServer is a follower, updates the isr in ZK and notifies the current leader.
+        replicaStateMachine.handleStateChanges(replicasFollowedByServer, OfflineReplica);
+
+        response.addAllRemainingLeaderBuckets(
```

Comment:
Add comment for this code to explain why we get the buckets lead by the server again, instead of using the above `bucketsLedByServer`.